Panic

```
cnf := &config.Config{
		Username: "qweqwe",
		Password: "qweqwe",
	}

	ctx := context.Background()

	ig, err := instagram.New(cnf)
	if err != nil {
		log.Fatal(err)
	}

	if err := ig.Login(ctx); err !=nil {
		log.Fatal(err)
	}

	Aresp, err := ig.Timeline().Feed(ctx, &instagram.TimelineFeedRequest{ UserID: 123})
	if err != nil {
		log.Fatal(err)
	}

	fmt.Println(Aresp.Items[0].MediaOrAd)
```


panic: runtime error: index out of range

--- a/src/timeline.go
+++ b/src/timeline.go
@@ -24,7 +24,6 @@ func (timeline *timeline) Feed(ctx context.Context, req *TimelineFeedRequest) (*
 	}
 
 	urlStru, _ := url.Parse(constants.TimelineFeedEndpoint) // TODO: handle error
-	query := urlStru.Query()
 
 	internalReq := &protos.TimelineFeedRequest{
 		UserID:    req.UserID,
@@ -32,14 +31,7 @@ func (timeline *timeline) Feed(ctx context.Context, req *TimelineFeedRequest) (*
 		RankToken: timeline.authManager.GenerateRankToken(req.UserID),
 	}
 
-	if internalReq.MaxID != "" {
-		query.Set("max_id", internalReq.MaxID)
-	}
-
-	urlStru.RawQuery = query.Encode()
-
-	_, body, _ := timeline.requestManager.Get(ctx, urlStru.String()) // TODO: handle error
-
+	_, body, _ := timeline.requestManager.Post(ctx, urlStru.String(), internalReq) // TODO: handle error
 	result := &protos.TimelineFeedResponse{}
 	json.Unmarshal([]byte(body), result) // TODO: handle error
 	return result, nil

--- a/src/timeline_test.go
+++ b/src/timeline_test.go
@@ -70,7 +70,7 @@ var _ = Describe("timeline", func() {
 
 		JustBeforeEach(func() {
 			mockAuthManager.On("GenerateRankToken", mock.Anything).Return(mockGenerateRankTokenResp)
-			mockRequestManager.On("Get", mock.Anything, mock.Anything).Return(nil, mockBody, nil)
+			mockRequestManager.On("Post", mock.Anything, mock.Anything, mock.Anything).Return(nil, mockBody, nil)
 
 			resp, err = client.Feed(ctx, req)
 
@@ -90,9 +90,9 @@ var _ = Describe("timeline", func() {
 				mockAuthManager.AssertCalled(GinkgoT(), "GenerateRankToken", req.UserID)
 			})
 
-			It("should call requestManager.Get", func() {
-				mockRequestManager.AssertNumberOfCalls(GinkgoT(), "Get", 1)
-				mockRequestManager.AssertCalled(GinkgoT(), "Get", mock.Anything, expURLStr)
+			It("should call requestManager.Post", func() {
+				mockRequestManager.AssertNumberOfCalls(GinkgoT(), "Post", 1)
+				mockRequestManager.AssertCalled(GinkgoT(), "Post", mock.Anything, expURLStr, mock.Anything)
 			})
 		})
 
@@ -111,12 +111,18 @@ var _ = Describe("timeline", func() {
 		Context("when MaxID is provided", func() {
 			BeforeEach(func() {
 				req.MaxID = "max id"
-
-				expURLQuery.Set("max_id", req.MaxID)
 			})
 
-			It("should add it to query string", func() {
-				mockRequestManager.AssertCalled(GinkgoT(), "Get", mock.Anything, expURLStr)
+			It("should add it to request body", func() {
+				mockRequestManager.AssertCalled(GinkgoT(), "Post",
+					mock.Anything,
+					mock.Anything,
+					mock.MatchedBy(func(internalReq *protos.TimelineFeedRequest) bool {
+						Expect(internalReq.MaxID).To(Equal(req.MaxID))
+
+						return true
+					}),
+				)
 			})
 		})
 	})

